CI: Conformance GKE fails Create: ResponseError: code=400, message=Resource_labels.value must be less than 63 bytes.

## CI failure

PR: https://github.com/cilium/cilium/pull/40250
Example failure: https://github.com/cilium/cilium/actions/runs/15931920018/job/44945387001?pr=40250#step:11:55

```
Run gcloud container clusters create cilium-cilium-15931920018-3-1 \
  gcloud container clusters create cilium-cilium-15931920018-3-1 \
    --labels "usage=cilium-cilium,owner=renovate-main-go-github-com-go-viper-mapstructure-v2-vulnerability" \
    --zone us-east1-c \
    --cluster-version 1.33 \
    --enable-ip-alias \
    --create-subnetwork="range=/26" \
    --cluster-ipv4-cidr="/21" \
    --services-ipv4-cidr="/24" \
    --image-type COS_CONTAINERD \
    --num-nodes 2 \
    --machine-type e2-custom-2-4096 \
    --disk-type pd-standard \
    --disk-size 20GB \
    --node-taints ignore-taint.cluster-autoscaler.kubernetes.io/cilium-agent-not-ready=true:NoExecute \
    --no-enable-insecure-kubelet-readonly-port
  
  native_cidr="$(gcloud container clusters describe cilium-cilium-15931920018-3-1 --zone us-east1-c --format 'value(clusterIpv4Cidr)')"
  echo native_cidr=${native_cidr} >> $GITHUB_OUTPUT
  shell: /usr/bin/bash -e {0}
  env:
    clusterName: cilium-cilium-15931920018-3
    USE_GKE_GCLOUD_AUTH_PLUGIN: true
    gcloud_version: 526.0.1
    job_name: Installation and Connectivity Test
    QUAY_ORGANIZATION: cilium
    QUAY_ORGANIZATION_DEV: cilium
    QUAY_CHARTS_ORGANIZATION_DEV: cilium-charts-dev
    EGRESS_GATEWAY_HELM_VALUES: --helm-set=egressGateway.enabled=true
    BGP_CONTROL_PLANE_HELM_VALUES: --helm-set=bgpControlPlane.enabled=true
    CILIUM_CLI_RELEASE_REPO: cilium/cilium-cli
    CILIUM_CLI_VERSION: 
    CILIUM_CLI_IMAGE_REPO: quay.io/cilium/cilium-cli-ci
    CILIUM_CLI_SKIP_BUILD: true
    CILIUM_CLI_CODE_OWNERS_PATHS: CODEOWNERS
    CILIUM_CLI_EXCLUDE_OWNERS: @cilium/github-sec
    PUSH_TO_DOCKER_HUB: true
    GCP_PERF_RESULTS_BUCKET: gs://cilium-scale-results
    CILIUM_RUNTIME_IMAGE_PREFIX: quay.io/cilium/
    KIND_VERSION: v0.29.0
    KIND_K8S_IMAGE: quay.io/cilium/kindest-node:v1.33.0@sha256:03032c2e474eef9d706416a077f2eeb14600653eecbdfada9fc2d81a4e0461f9
    KIND_K8S_VERSION: v1.33.0
    CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE: /home/runner/work/cilium/cilium/gha-creds-eb4103baf9b3ed0a.json
    GOOGLE_APPLICATION_CREDENTIALS: /home/runner/work/cilium/cilium/gha-creds-eb4103baf9b3ed0a.json
    GOOGLE_GHA_CREDS_PATH: /home/runner/work/cilium/cilium/gha-creds-eb4103baf9b3ed0a.json
    CLOUDSDK_CORE_PROJECT: ***
    CLOUDSDK_PROJECT: ***
    GCLOUD_PROJECT: ***
    GCP_PROJECT: ***
    GOOGLE_CLOUD_PROJECT: ***
    CLOUDSDK_METRICS_ENVIRONMENT: github-actions-setup-gcloud
    CLOUDSDK_METRICS_ENVIRONMENT_VERSION: 2.1.4
Note: The Kubelet readonly port (10255) is now deprecated. Please update your workloads to use the recommended alternatives. See https://cloud.google.com/kubernetes-engine/docs/how-to/disable-kubelet-readonly-port for ways to check usage and for migration instructions.
ERROR: (gcloud.container.clusters.create) ResponseError: code=400, message=Resource_labels.value must be less than 63 bytes.
Error: Process completed with exit code 1.
```

--- a/.github/actions/truncate-label/action.yml
+++ b/.github/actions/truncate-label/action.yml
@@ -1,0 +1,35 @@
+name: Truncate Label for Cloud Provider
+description: Truncates labels to meet cloud provider constraints (GKE 63-byte limit)
+inputs:
+  label:
+    description: 'The label value to truncate'
+    required: true
+  max_length:
+    description: 'Maximum length for the label (default 50 to allow for hash suffix)'
+    required: false
+    default: '50'
+outputs:
+  truncated_label:
+    description: 'The truncated label with hash suffix if needed'
+    value: ${{ steps.truncate.outputs.result }}
+runs:
+  using: composite
+  steps:
+    - name: Truncate label
+      id: truncate
+      shell: bash
+      run: |
+        LABEL="${{ inputs.label }}"
+        MAX_LENGTH="${{ inputs.max_length }}"
+
+        # Replace dots and slashes with hyphens for cloud provider compatibility
+        LABEL="${LABEL//[.\/]/-}"
+
+        # Truncate if needed and add hash for uniqueness
+        if [ ${#LABEL} -gt $MAX_LENGTH ]; then
+          HASH=$(echo "$LABEL" | sha256sum | cut -c1-8)
+          LABEL="${LABEL:0:$MAX_LENGTH}-${HASH}"
+        fi
+
+        echo "result=${LABEL}" >> $GITHUB_OUTPUT
+        echo "Truncated label: ${LABEL}"

--- a/.github/workflows/conformance-gke.yaml
+++ b/.github/workflows/conformance-gke.yaml
@@ -228,15 +228,16 @@ jobs:
           image-tag: ${{ inputs.SHA || github.sha }}
           chart-dir: ./untrusted/install/kubernetes/cilium
 
+      - name: Truncate owner label for GKE
+        id: truncate-owner
+        uses: ./.github/actions/truncate-label
+        with:
+          label: ${{ github.event_name == 'workflow_dispatch' && inputs.PR-number || github.ref_name }}
+
       - name: Set up job variables
         id: vars
         run: |
-          if [ "${{ github.event_name }}" = "workflow_dispatch" ]; then
-            OWNER="${{ inputs.PR-number }}"
-          else
-            OWNER="${{ github.ref_name }}"
-            OWNER="${OWNER//[.\/]/-}"
-          fi
+          OWNER="${{ steps.truncate-owner.outputs.truncated_label }}"
 
           CILIUM_INSTALL_DEFAULTS="${{ steps.default_vars.outputs.cilium_install_defaults }} \
             --helm-set=cluster.name=${{ env.clusterName }}-${{ matrix.config.index }} \

--- a/.github/workflows/net-perf-gke.yaml
+++ b/.github/workflows/net-perf-gke.yaml
@@ -170,16 +170,16 @@ jobs:
           chart-dir: ./install/kubernetes/cilium
           debug: false
 
+      - name: Truncate owner label for GKE
+        id: truncate-owner
+        uses: ./.github/actions/truncate-label
+        with:
+          label: ${{ (github.event_name == 'workflow_dispatch' || github.event.pull_request) && inputs.PR-number || github.ref_name }}
+
       - name: Set up job variables
         id: vars
         run: |
-          # shellcheck disable=SC2078
-          if [ "${{ github.event_name }}" = "workflow_dispatch" ] || [ ${{ github.event.pull_request }} ] ; then
-            OWNER="${{ inputs.PR-number }}"
-          else
-            OWNER="${{ github.ref_name }}"
-            OWNER="${OWNER//[.\/]/-}"
-          fi
+          OWNER="${{ steps.truncate-owner.outputs.truncated_label }}"
 
           CILIUM_INSTALL_DEFAULTS="${{ steps.default_vars.outputs.cilium_install_defaults }} \
             --helm-set=cluster.name=${{ env.clusterName }}-${{ matrix.index }} \

